student-store-ui/src/components/ProductCard/ProductCard.jsx

```diff
@@ -0,0 +1,73 @@
+import * as React from "react";
+import "./ProductCard.css";
+import { Link } from "react-router-dom";
+import { useState, useEffect } from "react";
+
+export default function ProductCard(props) {
+  const [quantity, setQuantity] = useState(0);
+
+  function addHandleQuantity(event) {
+    setQuantity((quantity) => quantity + 1);
+  }
+
+  function subHandleQuantity(event) {
+    setQuantity((quantity) => quantity - 1);
+  }
+
+  useEffect(() => {
+    if (props.shoppingCart.length === 0) {
+      setQuantity(0);
+    }
+  }, [props.shoppingCart]);
+
+  useEffect(() => {
+    let index = props.shoppingCart?.findIndex((cart) => {
+      return cart.product === props.product.name;
+    });
+    if (quantity === 1 && index === -1) {
+      const newCart = [
+        ...props.shoppingCart,
+        {
+          product: props.product.name,
+          quantity: quantity,
+          unit_price: props.product.price,
+          cost: props.product.price,
+        },
+      ];
+      props.setShoppingCart(newCart);
+    } else if (quantity >= 1) {
+      props.setShoppingCart((prevCart) => {
+        const updatedCart = [...prevCart];
+        updatedCart[index].quantity = quantity;
+        updatedCart[index].cost = props.product.price * quantity;
+
+        return updatedCart;
+      });
+    } else if (quantity === 0) {
+      props.setShoppingCart((shoppingCart) => shoppingCart.toSpliced(index, 1));
+    }
+  }, [quantity]);
+
+  console.log(props.shoppingCart);
```

Comment:
use helper functions/comments to make code more readable and remove all `console.log()` before submission